### PR TITLE
[FLINK-26747][flink-external-resource-gpu][JUnit5 Migration] Module: flink-external-resources

### PR DESCRIPTION
**Pull request by RocMarshal**
## What is the purpose of the change

[JUnit5 Migration] Module: flink-external-resources

## Brief change log

- org.apache.flink.externalresource.gpu.GPUDriverTest
- org.apache.flink.externalresource.gpu.GPUDiscoveryScriptTest

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / no / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by RocMarshal** [comment]
Hi, @wuchong @RyanSkraba , Could you please help to review this PR ? Thanks.

File: flink-external-resources/flink-external-resource-gpu/src/test/java/org/apache/flink/externalresource/gpu/GPUDiscoveryScriptTest.java [hunk 1]
```diff
@@ -19,41 +19,40 @@
...
-public class GPUDiscoveryScriptTest extends TestLogger {
+class GPUDiscoveryScriptTest {
 
     private static final String TEST_SCRIPT_PATH = "src/test/resources/test-coordination-mode.sh";
 
     @Test
-    public void testNonCoordinationMode() throws Exception {
-        assumeTrue(OperatingSystem.isLinux());
+    void testNonCoordinationMode() throws Exception {
+        assertThat(OperatingSystem.isLinux()).isTrue();
```
Comment (zentol) [inline flink-external-resources/flink-external-resource-gpu/src/test/java/org/apache/flink/externalresource/gpu/GPUDiscoveryScriptTest.java:37]: this should still use assume.

Comment (RocMarshal) [reply flink-external-resources/flink-external-resource-gpu/src/test/java/org/apache/flink/externalresource/gpu/GPUDiscoveryScriptTest.java:37]: @zentol  Thank you very much for the good catch. I'm so sorry for the rookie mistake. I'll check it carefully

Comment (RocMarshal) [reply flink-external-resources/flink-external-resource-gpu/src/test/java/org/apache/flink/externalresource/gpu/GPUDiscoveryScriptTest.java]: Done.

**Pull request merged**